### PR TITLE
experimental: add max chunk size parameter in SemanticChunker

### PR DESCRIPTION
**Pull request by killian-mahe**
**Description:** Add the possibility to limit the size of each chunk created with SemanticChunker.
**Issue:** https://github.com/langchain-ai/langchain/discussions/18014
**Dependencies:** None
**Twitter handle:** Don't have a Twitter account, but a [LinkedIn profil](https://www.linkedin.com/in/killian-mah%C3%A9-246928135/)

### CODE REVIEW

**Comment by hwchase17** [comment]
@killian-mahe cool idea! there are a bunch of merge conflicts since we updated this to support other methods - mind resolving them?

**Comment by killian-mahe** [comment]
> @killian-mahe cool idea! there are a bunch of merge conflicts since we updated this to support other methods - mind resolving them?

I'm glad this feature is appreciated, I'll work on this pull request and try to resolve conflicts today !

File: libs/experimental/langchain_experimental/text_splitter.py [hunk 1]
```diff
@@ -83,6 +94,71 @@ def calculate_cosine_distances(sentences: List[dict]) -> Tuple[List[float], List
...
+
+    return res
+
+
+def add_chunk(
+    sentences: List[dict],
+    start_index: int,
+    end_index: int,
+    chunks: List[str],
+    max_chunk_size: Union[str, None] = None,
```
Comment (NotThorz) [inline libs/experimental/langchain_experimental/text_splitter.py:135]: shouldn't this be an `Optional[int]` .

**Comment by infinite007** [comment]
Any update on this?

**Comment by thorgexyz** [comment]
hey this doesn't seem to be included in the latest version, or?

**Comment by cbarkinozer** [comment]
This would be helpful because SemanticChunker gets max lengtherror  with MilvusDB and Huggingface embedder.

```
ERROR:pymilvus.decorators:RPC error: [batch_insert], <ParamError: (code=1, message=invalid input, length of string exceeds max length. length: 93044, max length: 65535)>, <Time:{'RPC start': '2024-07-10 21:24:49.372050', 'RPC error': '2024-07-10 21:24:49.372050'}"
```

**Comment by ramkrishna757575** [comment]
Please add this feature because otherwise it is very difficult to use SemanticChunker.
Most of the times, it it creating an extremely big document chunk due to which context window is exceeded.

**Pull request closed without merging**